dora/src/semck/returnck.rs

```diff
@@ -58,9 +58,8 @@ impl<'a, 'ast> Visitor<'ast> for ReturnCheck<'a, 'ast> {
 pub fn returns_value(s: &Stmt) -> Result<(), Position> {
     match *s {
         StmtReturn(_) => Ok(()),
-        StmtLoop(ref stmt) => returns_value(&stmt.block),
         StmtFor(ref stmt) => Err(stmt.pos),
-        StmtWhile(ref stmt) => Err(stmt.pos),
+        StmtWhile(ref stmt) => returns_value(&stmt.block),
```

Comment:
While this worked for `loop`, it doesn't work for `while` since the body might not be run. Just return `Err(stmt.pos)` here.

Reply:
> While this worked for loop, it doesn't work for while since the body might not be run.

It didn't work for `loop` either, see the code example. I'd argue that it should be made to work for `while`, just like it should have been fixed for `loop` if `loop` wasn't dropped.

> Just return Err(stmt.pos) here.

Will do!

Reply:
> Just return Err(stmt.pos) here.

Done!

Reply:
> It didn't work for `loop` either, see the code example. I'd argue that it should be made to work for `while` (require `return` after `while`), just like it should have been fixed for `loop` if `loop` wasn't dropped.
Well there was a bug in this analysis with loop, with `while` this isn't even allowed.

Reply:
I'd argue that _allowed & works_ > _allowed & buggy_ > _not allowed_.

Nevermind, let's deal with it step by step. :-)